packages/components/src/components/segmented-control/segmented-control/segmented-control-styles.ts

```diff
@@ -1,18 +1,22 @@
-import { getCss } from '../../../utils';
+import { buildResponsiveStyles, getCss } from '../../../utils';
 import { addImportantToEachRule, hostHiddenStyles } from '../../../styles';
+import type { BreakpointCustomizable } from '../../../types';
 
 const MIN_ITEM_WIDTH = 46;
 const MAX_ITEM_WIDTH = 220;
 
-export const getComponentCss = (maxWidth: number): string => {
+export const getComponentCss = (maxWidth: number, column: BreakpointCustomizable<number | 'auto'>): string => {
   maxWidth = (maxWidth > MAX_ITEM_WIDTH && MAX_ITEM_WIDTH) || (maxWidth < MIN_ITEM_WIDTH && MIN_ITEM_WIDTH) || maxWidth;
 
   return getCss({
     '@global': {
       ':host': addImportantToEachRule({
         display: 'grid',
         gridAutoRows: '1fr', // for equal height
-        gridTemplateColumns: `repeat(auto-fit, ${maxWidth}px)`,
+        ...buildResponsiveStyles(column, (col: number | 'auto') => ({
+          gridTemplateColumns:
+            col === 'auto' ? `repeat(auto-fit, minmax(${maxWidth}px, 1fr))` : `repeat(${col}, ${maxWidth}px)`,
```

Comment:
Isn't this a regression or change in the old behavior? Before we didn't work with `minmax`.
Also should maxWidth really be involved when someone says I want 2 columns on a large screen?